src/test/java/org/javarosa/xform/parse/EntitiesTest.java

```diff
@@ -0,0 +1,351 @@
+                            t("name"),
+                            t("orx:meta",
+                                t("entities:entity dataset=\"people\"",
+                                    t("entities:create")
+                                )
+                            )
+                        )
+                    ),
+                    bind("/data/name").type("string").withAttribute("entities", "saveto", "name")
+                )
+            ),
+            body(
+                input("/data/name")
+            )
+        ));
+
+        scenario.getFormEntryController().addPostProcessor(new EntityFormPostProcessor());
+
+        Scenario deserializedScenario = scenario.serializeAndDeserializeForm();
+        deserializedScenario.getFormEntryController().addPostProcessor(new EntityFormPostProcessor());
+
+        deserializedScenario.next();
+        deserializedScenario.answer("Shiv Roy");
+
+        deserializedScenario.finalizeInstance();
+        List<Entity> entities = deserializedScenario.getFormEntryController().getModel().getAttachment(EntitiesAttachment.class).getEntities();
+        assertThat(entities.size(), equalTo(1));
+        assertThat(entities.get(0).dataset, equalTo("people"));
+        assertThat(entities.get(0).fields, equalTo(asList(new Pair<>("name", "Shiv Roy"))));
+    }
+
+    @Test
+    public void entitiesNamespaceWorksRegardlessOfName() throws IOException, DeserializationException {
+        Scenario scenario = Scenario.init("Create entity form", XFormsElement.html(
+            asList(
+                new Pair<>("blah", "http://www.opendatakit.org/xforms/entities")
+            ),
+            head(
+                title("Create entity form"),
+                model(
+                    mainInstance(
+                        t("data id=\"create-entity-form\"",
+                            t("name"),
+                            t("orx:meta",
+                                t("blah:entity dataset=\"people\"",
+                                    t("blah:create")
+                                )
+                            )
+                        )
+                    ),
+                    bind("/data/name").type("string").withAttribute("blah", "saveto", "name")
+                )
+            ),
+            body(
+                input("/data/name")
+            )
+        ));
+
+        scenario.getFormEntryController().addPostProcessor(new EntityFormPostProcessor());
+
+        scenario.next();
+        scenario.answer("Tom Wambsgans");
+
+        scenario.finalizeInstance();
+        List<Entity> entities = scenario.getFormEntryController().getModel().getAttachment(EntitiesAttachment.class).getEntities();
+        assertThat(entities.size(), equalTo(1));
+        assertThat(entities.get(0).fields, equalTo(asList(new Pair<>("name", "Tom Wambsgans"))));
+    }
+
+    @Test
+    public void mustUseCorrectNamespace() throws IOException {
```

Comment:
Ah, I see, it's only checked on create. So still two places to add and maybe the test should be split.

Reply:
Yeah, I think we should look at unit tests for `EntityPostProcessor` for this (or pull out something else that parses out the dataset/create). Coming back to this, I think we should do that now rather than file issues.